lib/services/LeaderElectionService.ts

```diff
@@ -0,0 +1,90 @@
+/*!
+ * Copyright (c) 2015-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * 
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+
+import { ServiceInterface, ServiceManagerOptions } from '../types';
+import {
+  BroadcastChannel,
+  createLeaderElection,
+  LeaderElector
+} from 'broadcast-channel';
+import { isBrowser } from '../features';
+
+declare type OnLeaderHandler = (() => void);
+declare type ServiceOptions = ServiceManagerOptions & {
+  onLeader?: OnLeaderHandler;
+};
+
+export class LeaderElectionService implements ServiceInterface {
```

Comment:
What is the motivation for moving the tab elector logic to it's own service?

Reply:
I isolated election-related code from ServiceManager into a separate class for single responsibility, then saw it looks like a service.